templates/masterclasses.html

```diff
@@ -12,6 +12,13 @@ <h1 class="">Masterclasses</h1>
           Register your session
         </a>
       </p>
+      <form class="p-search-box">
```

Comment:
Maybe consult a designer where to place the search input? I would expect it in the white area next to the "Filter previous sessions by topic". Also, we might want to only show some tags to filter by on the top initially and have an expand all link. So the tags don't take so much space and push the interesting content down.

Reply:
Yeah that's a good point. The Filters take a lot of space. Maybe we should move them somewhere else? That way we can add "Filter by Author/Duration" as well?

Reply:
As discussed in the meeting, piper suggested to put the searchbar closer to the cards. 

Other suggestions included adding an empty state for when there are no search results, replacing title "Previous sessions" with "Search results for ... ".

Reply:
Hello Aaryan,

Congrats on the work you've done so far, it looks like it's all coming together and that's very exciting! As we've discussed this morning, I'm adding my feedback regarding design below:

**Buttons & Interactive Elements**
- I believe registering sessions is the one of the main CTAs on the website considering the premise of the whole concept. Atm "Register your session" button for it looks like a secondary button. It would be better if it was a primary button (green) because it will be more prominent. The button styles are in the Design System Library in Figma. Lmk if you can't find it.
- I don't think the user needs to know how many talks there are related to that topic before filtering. Showing the number of talks in the filter button itself looks redundant to me. They can be removed, and it will look cleaner.
- When a filter is selected, the "Reset" option looks like a filter even though technically it's not. Because of that functionality difference it should be something less prominent i.g. a text link. I might have heard before somewhere that at Canonical we're against text links, so doesn't necessarily has to be that but you get the idea.
- Filter elements should have two-way interactions. Atm when the user selects a filter, they can't unselect by interacting with the same element. That is very unnatural.
- "Reset" option appears one line below the filter elements and pushes the content down. However with the current filters there is enough space to place it next to the last filter. That wouldn't break the page structure so it's a better approach. However, "Reset" would have to be placed the line below in the case the last line of filters finish exactly at the right border of that line. In the scenario that happens, there should be enough white space below the filters so that the element doesn't push the content below and break the page structure.

**Content & Page Structure**
- Filtering is above all sessions currently ( upcoming & past ) but it works only on the past sessions. Technically filtering should work on all content below which it is placed above. I would suggest to find a way to display upcoming sessions above the filter. Also this way, they will be more intriguing because it is the second content they see after the banner.
- If you go in this direction of categorizing content then you can have the title "Previous session" and above it display the filters. In this way you can also remove the "Filter previous sessions by topic" subtitle because it will be obvious that they are filtering that content.
- I think endless scroll is overwhelming. This is a personal preference of mine but I would much prefer if the content was paginated or displayed a certain amount of content with a button at the bottom labeled as "Show more".
- I don't know when the cover format for the talks was adapted but it makes the content look professional and cohesive. If possible, I would try to get a cover image for the talks that doesn't have it.

This is my two cents. Hope it helps :)

Cheers!

Reply:
I've noted down all of the comments and making individual issues. And resolving this since it doesn't contain specific recommendations for this PR.